.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,48 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Docker Build Test
+
+on:
+  workflow_dispatch: null
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  issues: read
+  pull-requests: read
+  actions: write
+
+jobs:
+  build:
+    name: Build for ${{ matrix.os }}
+    runs-on: ${{ matrix.runs_on }}
+    strategy:
+      matrix:
+        include:
+          - os: Linux
+            platform: linux
+            runs_on: ubuntu-latest
+            flags: --platform=linux/amd64
+          - os: Windows
+            platform: windows
+            runs_on: ubuntu-latest
+            flags: --platform=linux/amd64
+          - os: MacOS (Apple Silicon)
```

Comment:
Should we test the MacOS with Intel too?

Reply:
After further digging: Linux, Windows, and MacOS (Intel Based) can use amd64:

https://docs.docker.com/build/building/multi-platform/#:~:text=Many%20of%20the%20Docker%20Official%20Images%20available%20on%20Docker%20Hub%20support%20various%20architectures

Updating the action to be this:
```yml
jobs:
  build:
    name: Build for ${{ matrix.os }}
    runs-on: ${{ matrix.runs_on }}
    strategy:
      matrix:
        include:
          - os: Linux, Windows and MacOS x64
            platform: windows
            runs_on: ubuntu-latest
            flags: --platform=linux/amd64
          - os: MacOS (Apple Silicon)
            platform: macos
            runs_on: ubuntu-latest
            flags: --platform=linux/arm64

    steps:
      - uses: actions/checkout@v4
      - name: Docker Setup QEMU
        uses: docker/setup-qemu-action@v3.1.0
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3
      - name: Build the image
        run: |
          docker buildx build ${{ matrix.flags }} -t hyperjump/monika:${{ matrix.platform }} .
```

Reply:
I skipped the i386 (x32) because Docker will automatically run or build using x64

https://docs.docker.com/build/building/multi-platform/#:~:text=On%20an%20x86,invoke%20a%20build.